content/news/2022-01-06-bevy-0.6/index.md

```diff
@@ -0,0 +1,1007 @@
+#[derive(AsStd140)]
+pub struct MeshUniform {
+    pub transform: Mat4,
+    pub inverse_transpose_model: Mat4,
+}
+```
+
+Coupling an `AsStd140` derive with our new `UniformVec<T>` type makes it easy to write Rust types to shader-ready uniform buffers:
+
+```rust
+// WGSL shader
+struct Mesh {
+    model: mat4x4<f32>;
+    inverse_transpose_model: mat4x4<f32>;
+};
+
+[[group(2), binding(0)]]
+var<uniform> mesh: Mesh;
+```
+
+### UV Sphere Mesh Shape
+
+<div class="release-feature-authors">authors: @nside</div>
+
+Bevy now has a built-in "uv sphere" mesh primitive.
+
+```rust
+Mesh::from(UVSphere {
+    radius: 1.0,
+    sectors: 16,
+    stacks: 32,
+})
+```
+
+![uv sphere](uv_sphere.png)
+
+### Flat Normal Computation
+
+<div class="release-feature-authors">authors: @jakobhellermann</div>
+
+The `Mesh` type now has a `compute_flat_normals()` function. Imported GLTF meshes without normals now automatically have flat normals computed, [in accordance with the GLTF spec](https://www.khronos.org/registry/glTF/specs/2.0/glTF-2.0.html#meshes). 
+
+![flat normals](flat_normals.png)
+
+### Faster GLTF Loading
+
+<div class="release-feature-authors">authors: @DJMcNab, @mockers</div>
+
+@DJMcNab fixed nasty non-linear loading of GLTF nodes, which made them load much faster. One complicated scene went from 40 seconds to 0.2 seconds. Awesome!
+
+@mockersf made GLTF textures load asynchronously in Bevy's "IO task pool", which almost halved GLTF scene load times in some cases.
+
+## Bevy ECS
+
+### No more `.system()`!
+
+<div class="release-feature-authors">authors: @DJMcNab, @Ratysz</div>
+
+One of our highest priorities for Bevy ECS is "ergonomics". In the past I have made wild claims that Bevy ECS is the [most ergonomic ECS in existence](https://bevyengine.org/news/introducing-bevy/#ergonomics). We've spent gratuitous amounts of R&D pioneering new API techniques and I believe the results speak for themselves:
+
+```rust
+// This is a standalone Bevy 0.5 App that adds a simple `gravity` system to the App's schedule
+// and automatically runs it in parallel with other systems
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_system(gravity.system())
+        .run();
+}
+
+fn gravity(time: Res<Time>, mut query: Query<&mut Transform>) {
+    for mut transform in query.iter_mut() {
+        transform.translation.y += -9.8 * time.delta_seconds();
+    }
+}
+```
+
+I believe we were already the best in the market by a wide margin (especially if you take into account our automatic parallelization and change detection), but we had one thing holding us back from perfection ... that pesky `.system()`! We've tried removing it a number of times, but due to rustc limitations and safety issues, it always eluded us. Finally, @DJMcNab [found a solution](https://github.com/bevyengine/bevy/pull/2398). As a result, in Bevy 0.6 you can now register the system above like this:
+
+```rust
+// pure bliss!
+App::new()
+    .add_system(gravity)
```

Comment:
i think you should add `.add_plugins(DefaultPlugins)` too, currently it makes you think, that the `DefaultPlugins` are not needed any more.